src/mono/netcore/System.Private.CoreLib/src/System/Reflection/Emit/ConstructorOnTypeBuilderInst.cs

```diff
@@ -36,192 +36,225 @@
-					res [i] = RuntimeParameterInfo.New (parms [i], type, this, i + 1);
-				}
-			}
-			return res;
-		}
-
-		internal override Type[] GetParameterTypes () {
-			if (cb is ConstructorBuilder) {
-				return (cb as ConstructorBuilder).parameters;
-			} else {
-				ParameterInfo[] parms = cb.GetParameters ();
-				var res = new Type [parms.Length];
-				for (int i = 0; i < parms.Length; i++) {
-					res [i] = parms [i].ParameterType;
-				}
-				return res;
-			}
-		}
-
-		// Called from the runtime to return the corresponding finished ConstructorInfo object
-		internal ConstructorInfo RuntimeResolve () {
-			var type = instantiation.InternalResolve ();
-			return type.GetConstructor (cb);
-		}
-
-		public override int MetadataToken {
-			get {
-				return base.MetadataToken;
-			}
-		}
-
-		internal override int GetParametersCount ()
-		{
-			return cb.GetParametersCount ();
-		}
-
-		public override Object Invoke (Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
-		{
-			return cb.Invoke (obj, invokeAttr, binder, parameters,
-				culture);
-		}
-
-		public override RuntimeMethodHandle MethodHandle {
-			get {
-				return cb.MethodHandle;
-			}
-		}
-
-		public override MethodAttributes Attributes {
-			get {
-				return cb.Attributes;
-			}
-		}
-
-		public override CallingConventions CallingConvention {
-			get {
-				return cb.CallingConvention;
-			}
-		}
-
-		public override Type [] GetGenericArguments ()
-		{
-			return cb.GetGenericArguments ();
-		}
-
-		public override bool ContainsGenericParameters {
-			get {
-				return false;
-			}
-		}
-
-		public override bool IsGenericMethodDefinition {
-			get {
-				return false;
-			}
-		}
-
-		public override bool IsGenericMethod {
-			get {
-				return false;
-			}
-		}
-
-		//
-		// MethodBase members
-		//
-
-		public override object Invoke (BindingFlags invokeAttr, Binder binder, object[] parameters,
-									   CultureInfo culture)
-		{
-			throw new InvalidOperationException ();
-		}
-	}
+    /*
+     * This class represents a ctor of an instantiation of a generic type builder.
+     */
+    [StructLayout(LayoutKind.Sequential)]
+    internal class ConstructorOnTypeBuilderInst : ConstructorInfo
+    {
+        #region Keep in sync with object-internals.h
+        internal TypeBuilderInstantiation instantiation;
+        internal ConstructorInfo cb;
+        #endregion
+
+        public ConstructorOnTypeBuilderInst(TypeBuilderInstantiation instantiation, ConstructorInfo cb)
+        {
+            this.instantiation = instantiation;
+            this.cb = cb;
+        }
+
+        //
+        // MemberInfo members
+        //
+
+        public override Type DeclaringType
+        {
+            get
+            {
+                return instantiation;
+            }
+        }
+
+        public override string Name
+        {
+            get
+            {
+                return cb.Name;
```

Comment:
There's a lot of simple property getters that could become expression bodied. Are there fixers to do that?

Reply:
Yes, but I only fixed things for analyzers that we have set to produce warnings.